qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/utils/CDIUtils.java

```diff
@@ -63,4 +73,18 @@ public static String getSimpleName(String javaElementName, String annotationName
 		}
 		return javaElementName;
 	}
+
+	public static boolean isValidBean(IJavaElement javaElement) {
+		try {
+			return (javaElement.getClass().getDeclaringClass() == null

```

Comment:
I think `javaElement.getClass().getDeclaringClass()` would give you the declaring class for the instance of `IJavaElement` but not the instance represented in the code. `getClass()` comes from the Java language itself so it has no connection to JDT.